### PR TITLE
Implement Encode/Decode for BTreeSet 

### PR DESCRIPTION
**Pull request by satellitex**
I add Encode/Decode for BTreeSet, because it is not implement.
And `btree_set_workds` checked.

### CODE REVIEW

File: src/codec.rs [hunk 1]
```diff
@@ -943,6 +944,30 @@ impl<K: Decode + Ord, V: Decode> Decode for BTreeMap<K, V> {
 	}
 }
 
+impl<T: Encode + Ord> Encode for BTreeSet<T> {
+	fn encode_to<W: Output>(&self, dest: &mut W) {
+		let len = self.len();
+		assert!(len <= u32::max_value() as usize, "Attempted to serialize a collection with too many elements.");
+		(len as u32).encode_to(dest);
```
Comment (bkchr) [inline src/codec.rs]: Please use compact

Comment (satellitex) [reply src/codec.rs]: I Changed it.

**Comment by bkchr** [comment]
Can you please merge lasted master. We don't use option as result of decode anymore.

**Comment by satellitex** [comment]
@bkchr Many happened conflict. Can I resolve it?

**Comment by bkchr** [comment]
I don't understand your question.

**Comment by satellitex** [comment]
I execute `git merge origin/master` in this branch. 
And many conflicts have happened.

**Comment by satellitex** [comment]
Ah, The problem is that I forked from parity-codec-v3. I will fork from the master and try again.

**Comment by bkchr** [comment]
Please just cherry-pick your two commits into your branch.

**Comment by bkchr** [comment]
@pepyakin I don't understand your comment, the original set already should not contain any duplicates.

**Comment by cheme** [comment]
Not sure (not needed in most case), could be interesting to fail on unordered content when decoding (only allow set to be encoded/decoded as ordered content).

**Comment by bkchr** [comment]
Hmm yeah, but also adds extra complexity/cost to the decoding.

**Comment by satellitex** [comment]
@cheme I think that we should not allow using unordered contents sets.
Fortunately, Btree(Set|Map) meets that specification. But, Hash(Set|Map) is not.

**Comment by cheme** [comment]
@satellitex yes but as @bkchr point out, with current std btreeset implementation I do not know of a costless way to check content is rightly ordered without manually checking the ordering in the decode loop (checking if a value is added twice is natively supported but that will just do half the job of avoiding malleability).
This could be made optional, but it then does add upon code complexity.

**Comment by thiolliere** [comment]
Is unique Encoding/Decoding something we want to enforce ?

It seems to be an interesting property when we use it in the trie: two difference keys cannot be decoded as the same rust structure (or enum).

But I don't know how much it is interesting and important ?

**Comment by satellitex** [comment]
@thiolliere 
It is important to ensure that encoded a set of elements is always the same binary.
As you point out, there may be differences in the trie calculation results at each node.

**Comment by bkchr** [comment]
If someone sees any further issues or have other ideas for improvements, just open an issue :)

@cheme made decoding for the compact unique and the rest should already be unique.

**Pull request merged**